packages/sync-actions/src/utils/copy-empty-array-props.js

```diff
@@ -8,12 +8,25 @@
 export default function copyEmptyArrayProps(oldObj, newObj) {
   const newObjWithFixedEmptyArray = Object.entries(oldObj).reduce(
     (acc, [key, value]) => {
-      const replaceWithEmptyArray =
-        Array.isArray(value) && newObj[key] === undefined
-      acc[key] = replaceWithEmptyArray ? [] : newObj[key]
+      const isArray = Array.isArray(value)
+
+      // check if the value is array on `oldObj` and the key not found in `acc` (newObjWithFixedEmptyArray)
+      if (isArray && acc[key] === undefined) {
+        // init `key` with empty array
+        return { ...acc, [key]: [] }
+      }
+
+      // note that `typeof value === 'object'` returns true also in case of array
+      // check if its not array and its object
+      if (!isArray && typeof value === 'object' && acc[key]) {
```

Comment:
Just a question that came to my mind, do we ever have an array of objects here?

Reply:
nope, but i shall create new pull request with this feature, cause i wanted this PR to be simple.